### PR TITLE
[SYCLomatic] Remove requestFeature() call

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/ExprAnalysis.cpp [hunk 1]
```diff
@@ -1147,13 +1146,13 @@ void ExprAnalysis::analyzeType(TypeLoc TL, const Expr *CSCE) {
 
   auto Iter = MapNames::TypeNamesMap.find(TyName);
   if (Iter != MapNames::TypeNamesMap.end()) {
-    HelperFeatureSet.insert(Iter->second->RequestFeature);
-    requestHelperFeatureForTypeNames(TyName, SR.getBegin());
+    NeedDpctDeviceExtFlag = true;
```
Comment (tangjj11) [inline clang/lib/DPCT/ExprAnalysis.cpp:1149]: Should use `DpctGlobalInfo::setNeedDpctDeviceExt();`?

File: clang/lib/DPCT/Rules.h [hunk 2]
```diff
@@ -21,37 +21,37 @@ enum RulePriority { Takeover, Default, Fallback };
...
 struct TypeNameRule {
   std::string NewName;
-  clang::dpct::HelperFeatureEnum RequestFeature;
+  bool NeedDpctDeviceExtFlag;
   RulePriority Priority;
   std::vector<std::string> Includes;
   TypeNameRule(std::string Name)
-      : NewName(Name),
-        RequestFeature(clang::dpct::HelperFeatureEnum::no_feature_helper),
+      : NewName(Name), NeedDpctDeviceExtFlag(false),
```
Comment (tangjj11) [inline clang/lib/DPCT/Rules.h:28]: the default value is `true` may be more safety.

**Review by tangjj11 — APPROVED** [review]

File: clang/lib/DPCT/ExprAnalysis.cpp (hunk 1 above)
Comment (zhiweij1) [reply clang/lib/DPCT/ExprAnalysis.cpp:1149]: This `NeedDpctDeviceExtFlag` is a member in ExprAnalysis class, not the one in DpctGlobalInfo.
Current change just follows the original logic.

File: clang/lib/DPCT/Rules.h (hunk 2 above)
Comment (zhiweij1) [reply clang/lib/DPCT/Rules.h:28]: Maybe follow the original logic in this PR is better?

**Review by tomflinda — APPROVED** [review]
LGTM

**Pull request closed without merging**